### PR TITLE
[API-37990] add benefits documents flipper to POA v2 POST /2122 and /2122a

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

Add the `lighthouse_claims_api_poa_use_bd` feature flag to POA v2 POST /2122 and /2122a. When enabled, upload via Benefits Documents instead of VBMS (and vice versa).

## Related issue(s)

- [API-37990](https://jira.devops.va.gov/browse/API-37990)
- [API-37991](https://jira.devops.va.gov/browse/API-37991)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots
N/A

## What areas of the site does it impact?
Power of Attorney v2 POST /2122 requests

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Could use help testing end-to-end.

### CODE REVIEW

File: modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@ def perform(power_of_attorney_id, form_number = nil)
         power_of_attorney = ClaimsApi::PowerOfAttorney.find(power_of_attorney_id)
         rep_or_org = form_number == '2122A' ? 'representative' : 'serviceOrganization'
         poa_code = power_of_attorney.form_data&.dig(rep_or_org, 'poaCode')
-        doc_type = form_number == '2122A' ? 'L075' : 'L190'
+        doc_type = form_number == '2122' ? 'L190' : 'L075'
```
Comment (tycol7) [inline modules/claims_api/app/sidekiq/claims_api/v1/poa_form_builder_job.rb:23]: Needed to adjust v1 based on this [comment](https://jira.devops.va.gov/browse/API-37989?focusedId=3084428&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-3084428).

tl;dr - `L075` is more generic and applies to 2122 and 2122A, so if we don't have `form_number` explicitly defined, we fall back to `L075`.

**Comment by stiehlrod** [comment]
Tested v1
Screenshot 2024-07-31 at 8 20 04 AM

Now I am getting timeouts from BD
```2024-07-31 08:33:13.387245 I [61781:puma srv tp 004] Rails -- ClaimsApi :: benefits_documents :: {"detail":"/upload failure for claimId 3643de8f-c206-46b5-8831-b8dda6c36a6d, Gateway timeout"} :: Location: /Users/jennicastiehl/Oddball_Github/vets-api/modules/claims_api/lib/bd/bd.rb:61```

Will try again later, or someone else can pick up where I left off. We need to test POST V2 2122 individual & org

**Comment by stiehlrod** [comment]
> Will try again later, or someone else can pick up where I left off. We need to test POST V2 2122 individual & org

successfully uploaded a POA for a representative
Screenshot 2024-08-01 at 10 01 18 AM

Getting a gateway timeout for an individual
Screenshot 2024-08-01 at 10 04 57 AM

**Review by FonzMP — APPROVED** [review]
:shipit:
